### PR TITLE
fix: upgrade repo

### PR DESCRIPTION
**Pull request by nomego**
- upgrade vscode settings
- drop analysis.json, polymer.json
- switch to es-dev-server
- switch to karma testing
- upgrade all deps
- drop all wct deps
- add lit-html dev dep to avoid open-wc/testing error
- linting

TODO:
- convert the bdd-based tests (remaining .html)

Signed-off-by: Patrik Kullman <patrik.kullman@neovici.se>

### CODE REVIEW

**Comment by nomego** [comment]
codeclimate issue is the karma conf being too long, coverage has dropped since three of the test suites are bdd and not migrated, hopefully handled by @cristinecula along with cosmoz-sse tests

File: package.json [hunk 1]
```diff
@@ -61,40 +64,37 @@
...
+		"@neovici/eslint-config": "^1.2.0",
+		"@neovici/github-actions-eslint-annotator": "^0.4.0",
+		"@open-wc/testing": "^2.5.4",
+		"@open-wc/testing-karma": "^3.3.0",
 		"@polymer/iron-collapse": "^3.0.0",
 		"@polymer/iron-component-page": "^4.0.0",
 		"@polymer/iron-demo-helpers": "^3.0.0",
-		"@polymer/iron-list": "^3.0.0",
+		"@polymer/iron-list": "^3.1.0",
 		"@polymer/iron-test-helpers": "^3.0.0",
```
Comment (megheaiulian) [inline package.json]: Why are we keeping this ?

File: package.json [hunk 2]
```diff
@@ -61,40 +64,37 @@
...
-		"@polymer/paper-dropdown-menu": "^3.0.0",
-		"@polymer/paper-input": "^3.0.0",
+		"@polymer/paper-dropdown-menu": "^3.1.0",
+		"@polymer/paper-input": "^3.2.0",
 		"@polymer/paper-item": "^3.0.0",
 		"@polymer/paper-listbox": "^3.0.0",
 		"@polymer/paper-toggle-button": "^3.0.0",
 		"@polymer/test-fixture": "^4.0.2",
```
Comment (megheaiulian) [inline package.json]: Why are we keeping this ?

File: test/accordion.test.js [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+	setup(async () => {
+		tabs = await fixture(html`
+			<cosmoz-tabs accordion selected-values='["tab1","tab2"]'>
+				<cosmoz-tab name="tab0">Tab text 1</cosmoz-tab>
+				<cosmoz-tab name="tab1">Tab text 2</cosmoz-tab>
+				<cosmoz-tab name="tab2">Tab text 3</cosmoz-tab>
+			</cosmoz-tabs>
+		`);
+		const onIronItemsChanged = () => {
```
Comment (megheaiulian) [inline test/accordion.test.js]: We should use `oneEvent` from '@open-wc/testing' here.

File: test/basic.test.js [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+
+suite('defaults', () => {
+	let tabs;
+
+	suiteSetup(async () => {
+		await import('../cosmoz-tabs.js');
+	});
+
+	setup(async () => {
+		tabs = tabs = await fixture(html`
```
Comment (megheaiulian) [inline test/basic.test.js]: This looks wrong ? `tabs = tabs = ` ?

**Comment by cristinecula** [comment]
We should also include fix for https://github.com/Neovici/cosmoz-frontend/issues/2005

File: test/accordion.test.js (hunk 3 above)
Comment (nomego) [reply test/accordion.test.js]: Why are we even registering these event listeneres everywhere?

**Pull request merged**